srcpkgs/SDL/patches/SDL-1.2.15-sndio.patch

```diff
@@ -0,0 +1,436 @@
++		return(-1);
++	}
++
++	if (sio_getpar(hdl, &par) == 0) {
++		SDL_SetError("sio_getpar() failed");
++		return(-1);
++	}
++
++	if (par.bits == 16) {
++		if (par.sig && par.le) {
++			spec->format = AUDIO_S16LSB;
++		} else if (par.sig && !par.le) {
++			spec->format = AUDIO_S16MSB;
++		} else if (!par.sig && par.le) {
++			spec->format = AUDIO_U16LSB;
++		} else 
++			spec->format = AUDIO_U16MSB;
++	} else if (par.bits == 8) {
++		spec->format = par.sig ? AUDIO_S8 : AUDIO_U8;
++	} else {
++		SDL_SetError("SNDIO couldn't configure a suitable format");
++		return(-1);
++	}
++
++	spec->freq = par.rate;
++	spec->channels = par.pchan;
++	spec->samples = par.round;
++
++	SDL_CalculateAudioSpec(spec);
++
++	/* Allocate mixing buffer */
++	mixlen = spec->size;
++	mixbuf = (Uint8 *)SDL_AllocAudioMem(mixlen);
++	if ( mixbuf == NULL ) {
++		return(-1);
++	}
++	SDL_memset(mixbuf, spec->silence, spec->size);
++
++	if ( sio_start(hdl) == 0 ) {
++		SDL_SetError("sio_start() failed");
++		return(-1);
++	}
++
++	/* We're ready to rock and roll. :-) */
++	return(0);
++}
+diff -Naur SDL-1.2.15-orig/src/audio/sndio/SDL_sndioaudio.h SDL-1.2.15/src/audio/sndio/SDL_sndioaudio.h
+--- SDL-1.2.15-orig/src/audio/sndio/SDL_sndioaudio.h	1970-01-01 01:00:00.000000000 +0100
++++ SDL-1.2.15/src/audio/sndio/SDL_sndioaudio.h	2013-12-15 02:57:55.000000000 +0100
+@@ -0,0 +1,50 @@
++/*
++ * Copyright (c) 2008 Jacob Meuser <jakemsr@sdf.lonestar.org>
++ *
++ * Permission to use, copy, modify, and distribute this software for any
++ * purpose with or without fee is hereby granted, provided that the above
++ * copyright notice and this permission notice appear in all copies.
++ *
++ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
++ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
++ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
++ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
++ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
++ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
++ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
++ */
++
++#include "SDL_config.h"
++
++#ifndef _SDL_sndioaudio_h
++#define _SDL_sndioaudio_h
++
++#include <sndio.h>
++
++#include "../SDL_sysaudio.h"
++
++/* Hidden "this" pointer for the video functions */
++#define _THIS	SDL_AudioDevice *this
++
++struct SDL_PrivateAudioData {
++	/* The stream descriptor for the audio device */
++	struct sio_hdl *hdl;
++
++	/* The parent process id, to detect when application quits */
++	pid_t parent;
++
++	/* Raw mixing buffer */
++	Uint8 *mixbuf;
++	int    mixlen;
++
++};
++
++/* Old variable names */
++#define stream			(this->hidden->stream)
++#define parent			(this->hidden->parent)
++#define mixbuf			(this->hidden->mixbuf)
++#define mixlen			(this->hidden->mixlen)
++#define hdl			(this->hidden->hdl)
++
++#endif /* _SDL_sndioaudio_h */
++
```

Comment:
That's a massive patch. is there a chance for upstreaming sndio support?

Reply:
SNDIO support has been merged a long time ago but the SDL team never bothered to create a new stable 1.2 release. I could probably trim down this patch a little bit.